tests/ui/test_ui_tools.py

```diff
@@ -1001,6 +1001,42 @@ def test_topics_view(self, mocker, stream_button, width=40):
             for topic, count in zip(topic_list, unread_count_list)
         ])
 
+    @pytest.mark.parametrize('pinned', powerset([1, 2, 99, 1000]))
+    def test_stream_buttons(self, mocker, streams, pinned, width=40):
+        self.view.unpinned_streams = [s for s in streams if s[1] not in pinned]
+        self.view.pinned_streams = [s for s in streams if s[1] in pinned]
+
+        class button:
+            pass
+
+        def altbutton(stream, *args, **kwargs):
+            obj = button()
+            obj.stream_name = stream[0]
+            obj.stream_id = stream[1]
+            return obj
+
+        def altdivider(char):
+            obj = button()
+            return obj
+
+        def altlinebox(view, *args, **kwargs):
+            return view
+
+        def altstreamsview(btnlist, *args, **kwargs):
+            return btnlist
+
+        stream_button = mocker.patch(VIEWS + '.StreamButton',
+                                     side_effect=altbutton)
+        stream_view = mocker.patch(VIEWS + '.StreamsView',
+                                   side_effect=altstreamsview)
+        divider = mocker.patch(VIEWS + '.urwid.Divider',
+                               side_effect=altdivider)
```

Comment:
The above 3 mocks shouldn't be necessary, as we're wanting to test the actual code, not a mock. Also, for future reference, you don't test/use the mocks later, so you don't need to assign them to variables.